src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/BitSwapApiTest.cs

```diff
@@ -0,0 +1,528 @@
+            _dfsServiceOther = TestDfs.GetTestDfs(output, fileSystem2);
+        }
+
+        [Fact]
+        public async Task UnWant()
+        {
+            var fileSystem = Substitute.For<IFileSystem>();
+            fileSystem.GetCatalystDataDir().Returns(new DirectoryInfo(Path.Combine(Environment.CurrentDirectory,
+                $"dfs1-_{DateTime.Now:yyMMddHHmmssffff}")));
+
+            var dfsService = TestDfs.GetTestDfs(_testOutputHelper, fileSystem);
+            await dfsService.StartAsync();
+            
+            try
+            {
+                var cts = new CancellationTokenSource();
+                var block = new DagNode(Encoding.UTF8.GetBytes("BitSwapApiTest unknown block 2"));
+                Task wantTask = dfsService.BitSwapApi.GetAsync(block.Id, cts.Token);
+
+                var endTime = DateTime.Now.AddSeconds(10);
+                while (true)
+                {
+                    if (DateTime.Now > endTime)
+                    {
+                        throw new Xunit.Sdk.XunitException("wanted block is missing");
+                    }
+
+                    await Task.Delay(100, cts.Token);
+                    var w = await dfsService.BitSwapApi.WantsAsync(cancel: cts.Token);
+                    if (w.Contains(block.Id))
+                    {
+                        break;
+                    }
+                }
+
+                cts.Cancel();
+                dfsService.BitSwapApi.UnWant(block.Id, cts.Token);
+                var wants = await dfsService.BitSwapApi.WantsAsync(cancel: cts.Token);
+                wants.ToArray().Should().NotContain(block.Id);
+
+                //Race condition between threads
+                //Assert.True(wantTask.IsCanceled);
+            }
+            finally
+            {
+                await dfsService.StopAsync();
+            }
+        }
+        
+        [Fact]
+        public async Task Wants()
+        {
+            var fileSystem = Substitute.For<IFileSystem>();
+            fileSystem.GetCatalystDataDir().Returns(new DirectoryInfo(Path.Combine(Environment.CurrentDirectory,
+                $"dfs1-_{DateTime.Now:yyMMddHHmmssffff}")));
+            var dfsService = TestDfs.GetTestDfs(_testOutputHelper, fileSystem);
+            await dfsService.StartAsync();
+            try
+            {
+                var cts = new CancellationTokenSource();
+                var block = new DagNode(Encoding.UTF8.GetBytes("BitSwapApiTest unknown block"));
+                Task wantTask = dfsService.BitSwapApi.GetAsync(block.Id, cts.Token);
+
+                var endTime = DateTime.Now.AddSeconds(10);
+                while (true)
+                {
+                    if (DateTime.Now > endTime)
+                    {
+                        throw new Xunit.Sdk.XunitException("wanted block is missing");
+                    }
+                    
+                    await Task.Delay(100, cts.Token);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)